test/integration/com/unifina/service/PermissionServiceIntegrationSpec.groovy

```diff
@@ -103,6 +118,57 @@ class PermissionServiceIntegrationSpec extends IntegrationSpec {
 		uiChannelStream = new Stream(name: "ui channel", uiChannel: true, uiChannelCanvas: uiChannelCanvas, uiChannelPath: "/canvases/" + uiChannelCanvas.id + "/modules/2")
 		uiChannelStream.id = "stream-id-2"
 		uiChannelStream.save(validate: true, failOnError: true)
+
+		vulCan = new Canvas(name: "canvas")
+		vulCan.save(failOnError: true, validate: true)
+		category = new ModuleCategory(name: "category")
```

Comment:
`ModuleCategory` and `Module` are used to define the module registry, so I don't think you actually need to instantiate them here at all. The module ids in `Canvas#json`, `Stream#uiChannelPath` and `DashboardItem#module` doesn't refer to `Module#id`, instead they are just an identifier of an *instance* of a module within a canvas.

So in this test, as long as your `Stream#uiChannelPath` and `DashboardItem#module` refer to the same number (whatever number, you can still choose `1` for the allowed stream and `2` for the secret one), this test should work. So the creation of `ModuleCategory` and `Module` objects can be removed I think.